RELEASE/scripts/autoscend/auto_restore.ash

```diff
@@ -0,0 +1,991 @@
+	int goal = my_maxhp();
+	if(goalPercent > 1.0){
+		goal = ceil((goalPercent/100.0) * my_maxhp());
+	} else{
+		goal = ceil(goalPercent*my_maxhp());
+	}
+	return acquireHP(goal.to_int(), buyItems, useFreeRests);
+}
+
+/*
+ * Use a rest (can consume adventures if no free rests remain). Also attempts to maximize
+ * chateau bonus from resting if possible.
+ *
+ * TODO: if resting in campaway camp, check if we can/should use a burnt stick to get extra stats
+ *
+ * returns the number of times rested today (caller will have to work out if it rested or not)
+ */
+int doRest()
+{
+	if(chateaumantegna_available())
+	{
+		cli_execute("outfit save Backup");
+		chateaumantegna_nightstandSet();
+
+		boolean[item] restBonus = chateaumantegna_decorations();
+		stat bonus = $stat[none];
+		if(restBonus contains $item[Electric Muscle Stimulator])
+		{
+			bonus = $stat[Muscle];
+		}
+		else if(restBonus contains $item[Foreign Language Tapes])
+		{
+			bonus = $stat[Mysticality];
+		}
+		else if(restBonus contains $item[Bowl of Potpourri])
+		{
+			bonus = $stat[Moxie];
+		}
+
+		boolean closet = false;
+		item grab = $item[none];
+		item replace = $item[none];
+		switch(bonus)
+		{
+		case $stat[Muscle]:
+			replace = equipped_item($slot[off-hand]);
+			grab = $item[Fake Washboard];
+			if(can_equip($item[LOV Eardigan]) && (item_amount($item[LOV Eardigan]) > 0))
+			{
+				equip($slot[shirt], $item[LOV Eardigan]);
+			}
+			break;
+		case $stat[Mysticality]:
+			replace = equipped_item($slot[off-hand]);
+			grab = $item[Basaltamander Buckler];
+			if(can_equip($item[LOV Epaulettes]) && (item_amount($item[LOV Epaulettes]) > 0))
+			{
+				equip($slot[back], $item[LOV Epaulettes]);
+			}
+			break;
+		case $stat[Moxie]:
+			replace = equipped_item($slot[weapon]);
+			grab = $item[Backwoods Banjo];
+			if(can_equip($item[LOV Earrings]) && (item_amount($item[LOV Earrings]) > 0))
+			{
+				equip($slot[acc1], $item[LOV Earrings]);
+			}
+			break;
+		}
+
+		if((grab != $item[none]) && possessEquipment(grab) && (replace != grab) && can_equip(grab))
+		{
+			equip(grab);
+		}
+		if(!possessEquipment(grab) && (replace != grab) && (closet_amount(grab) > 0) && can_equip(grab))
+		{
+			closet = true;
+			take_closet(1, grab);
+			equip(grab);
+		}
+
+		visit_url("place.php?whichplace=chateau&action=chateau_restbox");
+
+		if((replace != grab) && (replace != $item[none]))
+		{
+			equip(replace);
+		}
+		cli_execute("outfit Backup");
+		if(closet)
+		{
+			if(item_amount(grab) > 0)
+			{
+				put_closet(1, grab);
+			}
+		}
+
+	}
+	else
+	{
+		set_property("restUsingChateau", false);
+		cli_execute("rest");
+		set_property("restUsingChateau", true);
+	}
+	return get_property("timesRested").to_int();
+}
+
+boolean haveFreeRestAvailable(){
+	return get_property("timesRested").to_int() < total_free_rests();
+}
+
+int freeRestsRemaining(){
+	return max(0, total_free_rests() - get_property("timesRested").to_int());
+}
+
+boolean haveAnyIotmAlternateCampsight(){
```

Comment:
s/Campsight/Campsite

I can't really imagine a case where I'd need this to be externally usable, but my lack of imagination doesn't mean it's useless.

Reply:
youre probably right. ill remove it from the public interface. I made this before i refactored the restore stuff in a single place and I was referencing it in both community service and util (and figured other paths might want to as well).